From 1dfb8242e8d982d036399766c4af62ddc221e38d Mon Sep 17 00:00:00 2001
From: Namjae Jeon <namjae.jeon@samsung.com>
Date: Fri, 18 Jun 2021 10:29:56 +0900
Subject: [PATCH] ksmbd: use goto instead of duplicating the resoure cleanup in
 ksmbd_open_fd
Git-commit: 1dfb8242e8d982d036399766c4af62ddc221e38d
Patch-mainline: v5.15-rc1
References: bsn#12
Modified-by-SEL: No


Use goto instead of duplicating the resoure cleanup in ksmbd_open_fd.

Reviewed-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/cifsd/vfs_cache.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/fs/cifsd/vfs_cache.c b/fs/cifsd/vfs_cache.c
index 3f18018668b6..71a11128d908 100644
--- a/fs/cifsd/vfs_cache.c
+++ b/fs/cifsd/vfs_cache.c
@@ -559,19 +559,22 @@ struct ksmbd_file *ksmbd_open_fd(struct ksmbd_work *work, struct file *filp)
 	fp->f_ci		= ksmbd_inode_get(fp);
 
 	if (!fp->f_ci) {
-		kmem_cache_free(filp_cache, fp);
-		return ERR_PTR(-ENOMEM);
+		ret = -ENOMEM;
+		goto err_out;
 	}
 
 	ret = __open_id(&work->sess->file_table, fp, OPEN_ID_TYPE_VOLATILE_ID);
 	if (ret) {
 		ksmbd_inode_put(fp->f_ci);
-		kmem_cache_free(filp_cache, fp);
-		return ERR_PTR(ret);
+		goto err_out;
 	}
 
 	atomic_inc(&work->conn->stats.open_files_count);
 	return fp;
+
+err_out:
+	kmem_cache_free(filp_cache, fp);
+	return ERR_PTR(ret);
 }
 
 static int
-- 
2.33.0

